apps/designer/app/designer/features/sidebar-left/panels/pages/pages.tsx

```diff
@@ -110,16 +137,44 @@ export const TabContent = (props: TabContentProps) => {
     navigate(`${props.config.designerPath}/${project.id}?pageId=${pageId}`);
   };
 
-  if (currentPageId === undefined) {
+  const [newPageOpen, setNewPageOpen] = useState(false);
+  const [newPageState, setNewPageState] = useState<Fetcher["state"]>("idle");
+
+  if (currentPageId === undefined || project === undefined) {
     return null;
   }
 
   return (
-    <PagesPanel
-      onClose={() => props.onSetActiveTab("none")}
-      onSelect={handleSelect}
-      selectedPageId={currentPageId}
-    />
+    <>
+      <PagesPanel
+        onClose={() => props.onSetActiveTab("none")}
+        onNewPage={() => {
+          if (newPageOpen === false) {
+            setNewPageOpen(true);
+          }
+          if (newPageOpen && newPageState === "idle") {
```

Comment:
is that preventing from closing the new page content when it's saving or loading? Should we be preventing it?

Reply:
> is that preventing from closing the new page content when it's saving or loading?

Yeah

> Should we be preventing it?

Do you think it's a bad idea? Form submition may fail, and if the panel is closed by that time user will not see the errors.

Reply:
Generally preventing user from opening/closing things and making them to wait is always bad.
In this case what we can do is:
1. indicate we are sending, so they know they may miss something if they close (it's an edge case anyways, it should be fast enough not to notice)
2. if its easy enough we can resort to a toast if user has already closed, but if it makes logic complicated - don't worry, its not a normal case

Important is that the user knows what's happening when sending the state.

Reply:
Yeah, `#2` will be too much complication. I've made it so the button says `Creating...` for now, maybe not ideal solution but this is not a final design anyway